src/main/encryptors/LibsodiumEncryptor.ts

```diff
@@ -28,51 +22,71 @@ export class LibsodiumEncryptor implements Encryptor {
   constructor(private props: LibsodiumEncryptorProps) {}
 
   async encrypt<T, U extends keyof T>(props: EncryptProps<T, U>): Promise<EncryptResult<T, U>> {
-    const { item, fieldsToEncrypt } = props
-    const key = await this.props.keyProvider.generateDataKey()
-    const nonce = randombytes_buf(crypto_secretbox_NONCEBYTES)
+    const { item } = props
+    const plaintextFields = { ...item } as Omit<T, U>
+    const fieldsToEncrypt = {} as Pick<T, U>
 
-    const encryptedFields: {
-      [K in U]: Uint8Array
-    } = {} as ItemWithEncryptedFields<T, U>
-
-    fieldsToEncrypt.forEach((fieldName) => {
+    props.fieldsToEncrypt.forEach((fieldName) => {
+      delete (plaintextFields as any)[fieldName]
       const fieldValue = item[fieldName]
-      if (fieldValue !== undefined && fieldValue !== null) {
-        encryptedFields[fieldName] = crypto_secretbox_easy(serialize(fieldValue), nonce, key.plaintextKey)
+      if (fieldValue !== undefined) {
+        fieldsToEncrypt[fieldName] = fieldValue
       }
     })
 
-    const encryptedItem = {
-      ...item,
-      ...encryptedFields,
-      __jayz__metadata: {
-        encryptedDataKey: key.encryptedKey,
-        keyId: key.metadata?.keyId,
-        nonce,
-        version: this.version,
-        encryptedFieldNames: fieldsToEncrypt
+    const nonce = randombytes_buf(crypto_secretbox_NONCEBYTES)
+    const key = await this.props.keyProvider.generateDataKey()
+    const encryptedFields = crypto_secretbox_easy(serialize(fieldsToEncrypt), nonce, key.plaintextKey)
+    return {
```

Comment:
We should `memzero` the plaintext keys in this class after use -- I'll submit that as a separate PR.